src/PostgREST/App.hs

```diff
@@ -227,23 +229,28 @@ app dbStructure conf apiRequest =
               let acceptH = (hAllow, if tableInsertable table then "GET,POST,PATCH,DELETE" else "GET") in
               return $ responseLBS status200 [allOrigins, acceptH] ""
 
-        (ActionInvoke, TargetProc qi, Just (PayloadJSON payload)) ->
+        -- These two are ensured by shouldParsePayload in ApiRequest
+        (ActionInvoke{isReadOnly=False}, _, Nothing) -> pure notFound
```

Comment:
What do these two notFound matchers do? When I remove them locally it still builds and the tests still pass.